Standards/scs-01XX-v1-secure-connections.md

```diff
@@ -0,0 +1,254 @@
+
+### Database Connections
+
+- The database servers used by the OpenStack services MUST be configured for TLS transport.
+- All OpenStack services MUST have TLS configured for the database connection via the `ssl_ca` directive. See [OpenStack service database configuration](https://docs.openstack.org/security-guide/databases/database-access-control.html#openstack-service-database-configuration).
+- Database user accounts for the OpenStack services SHOULD be configured to require TLS connections via the `REQUIRE SSL` SQL directive. See [Require user accounts to require SSL transport](https://docs.openstack.org/security-guide/databases/database-access-control.html#require-user-accounts-to-require-ssl-transport).
+- Security MAY be further enhanced by configuring the OpenStack services to use X.509 client certificates for database authentication. See [Authentication with X.509 certificates](https://docs.openstack.org/security-guide/databases/database-access-control.html#authentication-with-x-509-certificates).
+
+### Message Queue Connections
+
+- If using RabbitMQ or Qpid as the message queue service, the SSL functionality of the message broker MUST be enabled and used by the OpenStack services. See [Messaging transport security](https://docs.openstack.org/security-guide/messaging/security.html#messaging-transport-security).
+  - If using RabbitMQ, all OpenStack services' oslo.messaging configuration for RabbitMQ MUST specify options accordingly to enable SSL:
+    ```ini
+    [oslo_messaging_rabbit]
+    ssl = true
+    ssl_ca_file = /path/to/file
+    ssl_key_file =
+    ssl_cert_file =
+    ```
+    (`ssl_ca_file` MUST be set to the path of the CA certificate, `ssl_key_file` and `ssl_cert_file` for client certificates are OPTIONAL)
+  - If using Qpid, the OpenStack services' oslo.messaging configuration for Qpid MUST set the `qpid_protocol` option to `ssl` to enable SSL.
+- If using Apache Kafka, the server listener MUST be configured to accept SSL connections. See [Apache Kafka Listener Configuration](https://kafka.apache.org/documentation/#listener_configuration).
+  - The OpenStack services' oslo.messaging configuration for Kafka MUST specify `security_protocol` as either `SSL` or `SASL_SSL` and the related options appropriately. See [Kafka Driver Options](https://docs.openstack.org/oslo.messaging/latest/admin/kafka.html#driver-options).
+
+### Hypervisor and Live Migration Connections
+
+- If QEMU and libvirt are used as the hypervisor interface in Nova, QEMU-native TLS SHOULD be used. See [Secure live migration with QEMU-native TLS](https://docs.openstack.org/nova/latest/admin/secure-live-migration-with-qemu-native-tls.html).
```

Comment:
Did I understand it correctly, that in the Open Question section, you did not choose between QEMU-native TLS and SSH tunneling? Then why is only the first recommended here?

If QEMU-native TLS  is the better Option, it is not an Open Question anymore ;)

Reply:
There are actually two layers of this:

Layer 1: decide between a) libvirt TLS + QEMU native TLS migration or b) local libvirt UNIX socket + SSH migration. This decides the general direction of whether the libvirt port stays exposed or not.

Layer 2: when choosing either a) or b) above, there is one thing left to address: how to authorize/limit clients connecting to the interface (either via TCP+TLS or SSH).
Using a) or b) alone still poses a substantial threat due to the vast privileges granted upon connection (controlling the whole hypervisor having root privileges).

The section you  quoted aims to address layer 1 here because QEMU-native TLS is much better documented and also recommended by upstream.

The Open Questions section focuses more on the full picture including layer 2. When taking this into account, the choice between a) or b) is not as easy anymore. The QEMU-native TLS approach would require client certificate management which is not easy at all. So when considering layer 1 and 2, then it's still an open question imho.

I think I need to be clearer in the questions and/or decision sections. Thanks for bringing this up! It made me realize that this can be confusing.